XouDevSpec/Classes/features/Apis/XouBase/Service/XouBaseApiService.swift

```diff
@@ -0,0 +1,121 @@
+//
+//  XouBaseApiCalls.swift
+//  XouDevSpec
+//
+//  Created by Xander Schoeman on 2020/04/19.
+//
+import Foundation
+
+public struct XouBaseApiCalls: XouBaseApiServiceProtocol {
+    var urls = UrlStrings()
+
+    public init() {
+    }
+    
+    public  func getUsers(completetionHandler: @escaping(Result<[User], AnimeError>) -> Void) {
+        
+        
+        guard let url = URL(string: urls.getUsersLocalUrl) else {
+                return
+            }
+        
+            var request = URLRequest(url: url,
+                cachePolicy: .useProtocolCachePolicy,
+            timeoutInterval: 10.0)
+
+
+        request.httpMethod = "GET"
+
+        let session = URLSession.shared
+        let dataTask = session.dataTask(with: request as URLRequest) { data, _, _ in
+            guard let jsonData = data else {
+                completetionHandler(.failure(.noDataAvailable))
+                return
+            }
+            do {
+                let decoder = JSONDecoder()
+                let response = try decoder.decode([User].self, from: jsonData)
+                completetionHandler(.success(response))
+                
+            } catch {
+                completetionHandler(.failure(.cannotProcessData))
+            }
+        }
+
+        dataTask.resume()
+      }
+    
+        public  func registerUser(theUser: User,completetionHandler: @escaping(Result<User, AnimeError>) -> Void) {
+
+            do {
+                guard let url = URL(string: urls.registerUserUrl) else {
+                        return
+                    }
+                  
+                  var request = URLRequest(url: url)
+
+                  
+                request.httpMethod = "POST"
```

Comment:
A suggestion, consider putting "POST" and "GET" in a struct then you use that struct to get the value. To reduce chances of mistakes if changes are made

Reply:
On it.